File: src/main/java/com/gusrylmubarok/reddit/backend/repository/SubredditRepository.java
package com.gusrylmubarok.reddit.backend.repository;

import com.gusrylmubarok.reddit.backend.model.Subreddit;
import org.springframework.data.jpa.repository.JpaRepository;

import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface SubredditRepository extends JpaRepository<Subreddit, Long> {
    Optional<Subreddit> findByNameEquals(String subredditName);
}

File: src/main/kotlin/com/reddot/repository/SubredditRepository.kt
package com.reddot.repository

import com.reddot.data.entity.Subreddit
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface SubredditRepository : JpaRepository<Subreddit, Long> {
    fun findByName(subredditName: String): Optional<Subreddit>
}